filecoin/store/content.js

```diff
@@ -0,0 +1,60 @@
+import { StoreOperationFailed, RecordNotFound } from '@web3-storage/filecoin-api/errors'
+import pRetry from 'p-retry'
+
+/**
+ * @typedef {import('multiformats').UnknownLink} UnknownLink
+ * @typedef {import('@web3-storage/filecoin-api/storefront/api').ContentStore<UnknownLink, Uint8Array>} ContentStore
+ */
+
+/**
+ * @param {URL} storeHttpEndpoint
+ * @returns {ContentStore}
+ */
+export const useContentStore = (storeHttpEndpoint) => {
+  return {
+    /**
+     * Stream Blob bytes for a given CID.
+     *
+     * @param {import('@ucanto/interface').UnknownLink} cid 
+     */
+    stream: async (cid) => {
+      // create URL for the link to be fetched
+      const getUrl = new URL(`/${cid.toString()}`, storeHttpEndpoint)
+
+      // Retry a few times as it looks like R2 sometimes takes a bit to make it available
```

Comment:
context: This was making integration tests sometimes fail before I added the retry